microprofile/WFLY-19954_Preview_Support_vertx_feature_pack.adoc

```diff
@@ -0,0 +1,234 @@
+
+The feature pack will be integrated to `preview/feature-pack` in `Preview` stability level.
+
+==== Configure the {vertx_options_with_link} in vertx subsystem
+The {vertx_options_with_link} is used to customize the options to construct a Vertx instance. The following fields in primitive types are selected to configure as attributes in the vertx subsystem with address `/subsystem=vertx/vertx-option=xx`:
+
+.Attributes in address: `/subsystem=vertx/vertx-option=xxx`
+[frame=all, grid=all]
+|===
+^| Field Name ^| Data Type ^| Source Class ^| Resource Attribute Name ^| Note
+
+| eventLoopPoolSize | int | {vertx_options_with_link} | event-loop-pool-size | number of event loop threads to be used
+| workerPoolSize | int | {vertx_options_with_link} | worker-pool-size | number of threads in the worker pool
+| internalBlockingPoolSize | int | {vertx_options_with_link} | internal-blocking-pool-size | number of threads in the internal blocking pool
+| preferNativeTransport | boolean | {vertx_options_with_link} | prefer-native-transport | If native transport is preferred or not
+| blockedThreadCheckInterval | long | {vertx_options_with_link} | blocked-thread-check-interval | blocked thread check interval
+| blockedThreadCheckIntervalUnit | TimeUnit | {vertx_options_with_link} | blocked-thread-check-interval-unit | blocked thread check interval time unit
+| maxEventLoopExecuteTime | long | {vertx_options_with_link} | max-eventloop-execute-time | max event loop thread execution time
+| maxEventLoopExecuteTimeUnit | TimeUnit | {vertx_options_with_link} | max-eventloop-execute-time-unit | max event loop thread execution time unit
+| maxWorkerExecuteTime | long | {vertx_options_with_link} | max-worker-execute-time | max worker thread execution time
+| maxWorkerExecuteTimeUnit | TimeUnit | {vertx_options_with_link} | max-worker-execute-time-unit | max worker thread execution time unit
+| warningExceptionTime | long | {vertx_options_with_link} | warning-exception-time | max time a thread can be blocked before stack traces get logged
+| warningExceptionTimeUnit | TimeUnit | {vertx_options_with_link} | warning-exception-time-unit | the time unit for warningExceptionTime
+| disableTCCL | boolean | {vertx_options_with_link} | disable-tccl | whether Vert.x sets the Context classloader as the thread context classloader on actions executed on that Context.
+| useDaemonThread | int | {vertx_options_with_link} | use-daemon-thread | whether we want to use daemon vertx thread
+| classPathResolvingEnabled | boolean | {filesystem_options_with_link} | classpath-resolving-enabled | whether classpath resolving is enabled
+| fileCachingEnabled | boolean | {filesystem_options_with_link} | file-cache-enabled | whether file caching is enabled for class path resolving
+| fileCacheDir | String | {filesystem_options_with_link} | file-cache-dir | file cache directory
```

Comment:
The default value is `System.getProperty("java.io.tmpdir", ".") + "/vertx-cache"`, shall we update the default cache directory to `${jboss.server.temp.dir}/vertx-cache` ?

Reply:
Defaulting to the server default directory seems wise to me.